kogito-editors-js/packages/import-java-classes-component/src/components/ImportJavaClasses/ImportJavaClassesWizard.tsx

```diff
@@ -33,37 +37,70 @@ export const ImportJavaClassesWizard: React.FunctionComponent<ImportJavaClassesW
   buttonTooltipMessage,
 }: ImportJavaClassesWizardProps) => {
   const { i18n } = useImportJavaClassesWizardI18n();
-  const [selectedJavaClasses, setSelectedJavaClasses] = useState<string[]>([]);
+  const [javaClasses, setJavaClasses] = useState<JavaClass[]>([]);
   const updateSelectedClasses = (fullClassName: string, add: boolean) => {
     if (add) {
-      const classesSet = new Set(selectedJavaClasses);
-      classesSet.add(fullClassName);
-      setSelectedJavaClasses(Array.from(classesSet));
+      if (!javaClasses.some((javaClass) => javaClass.name === fullClassName)) {
```

Comment:
I think it's better to assign a variable to that `if` condition, and inline this nested `if` with the outer (by using `&&`) to reduce cyclomatic complexity.

Reply:
@vpellegrino But in that way the condition which leads to `else` branch will change (eg. If` add == true` and className is not present, the `else` branch will be called, while this should be used only in case` add = false`). I could add an `else if`, but I think it is more appropriate to separate logic based the `add` branch and the `remove` branch

Reply:
Ok, got it. So please wrap the inner code in a separate method.

Reply:
@vpellegrino done (y)